src/main/java/com/example/rebrain/services/CardService.java

```diff
@@ -1,35 +1,61 @@
 package com.example.rebrain.services;
 
-import com.example.rebrain.models.Card;
-import com.example.rebrain.repositories.CardRepository;
+import com.example.rebrain.exception.CardNotFoundException;
+import com.example.rebrain.entity.CardEntity;
+import com.example.rebrain.model.UpdateCard;
+import com.example.rebrain.repositories.CardRepo;
+import com.example.rebrain.model.Card;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
 public class CardService {
-    private final CardRepository cardRepository;
+    @Autowired
+    private final CardRepo cardRepo;
 
-    public List<Card> listCards(String title) {
-        if (title != null) return cardRepository.findByTitle(title);
-        return cardRepository.findAll();
+    public CardEntity create(CardEntity card) {
+        log.info("Saving new {}", card);
+        return cardRepo.save(card);
     }
 
-    public void saveCard(Card card) {
-        log.info("Saving new {}", card);
-        cardRepository.save(card);
+    public List<CardEntity> getAll() {
+        return cardRepo.findAll();
+    }
+
+    public CardEntity getOne(Integer id) throws CardNotFoundException {
```

Comment:
Such exceptions must be _unchecked_ because it is based on business functionality rather than technical